extensions/positron-python/python_files/posit/positron/convert.py

```diff
@@ -282,30 +393,38 @@ def convert_filters(self) -> Optional[str]:
         return None
 
     def _convert_between_filter(self) -> str:
-        assert isinstance(self.filter_key.params, FilterBetween)
-        is_between = self.filter_key.filter_type == RowFilterType.Between
-        left = self.filter_key.params.left_value
-        right = self.filter_key.params.right_value
-        operator = "" if is_between else "~"
-        return f"{operator}{self.table_name}{self.column_name}.between({left}, {right})"
+        raise NotImplementedError("Subclasses must implement _convert_between_filter method")
 
     def _convert_compare_filter(self) -> str:
-        """Handle comparison filters such as equals, not equals, greater than, etc.
+        raise NotImplementedError("Subclasses must implement _convert_compare_filter method")
 
-        Returns
-        -------
-        str
-            Filter comparison code string.
-        """
-        assert isinstance(self.filter_key.params, FilterComparison)
-        op = (
-            "=="
-            if self.filter_key.params.op.value == FilterComparisonOp.Eq
-            else self.filter_key.params.op.value
-        )
+    def _convert_text_search_filter(self) -> str:
+        raise NotImplementedError("Subclasses must implement _convert_text_search_filter method")
+
+    def _convert_is_empty_filter(self) -> str:
+        raise NotImplementedError("Subclasses must implement _convert_is_empty_filter method")
+
+    def _convert_not_empty_filter(self) -> str:
+        raise NotImplementedError("Subclasses must implement _convert_not_empty_filter method")
+
+    def _convert_is_null_filter(self) -> str:
+        raise NotImplementedError("Subclasses must implement _convert_is_null_filter method")
+
+    def _convert_not_null_filter(self) -> str:
+        raise NotImplementedError("Subclasses must implement _convert_not_null_filter method")
+
+    def _convert_is_true_filter(self) -> str:
+        raise NotImplementedError("Subclasses must implement _convert_is_true_filter method")
+
+    def _convert_is_false_filter(self) -> str:
+        raise NotImplementedError("Subclasses must implement _convert_is_false_filter method")
```

Comment:
You could also use `abc.abstractmethod` and make these method bodies `pass`

Reply:
oh, that is much nicer! updated 💯